OpenRA.Mods.Common/Traits/World/Pathfinder/PathFinderInfo.cs

```diff
@@ -0,0 +1,51 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Calculates routes for mobile units based on the A* search algorithm.", " Attach this to the world actor.")]
+	public class PathFinderInfo : ITraitInfo
```

Comment:
The problem are decorators. I'm using 2 implementations for the trait using the decorator design pattern, one for caching and one for actual calculations. Keeping them in the same file would make a quite monster file which, to be honest, makes it harder to understand and read. Do I really have to do that? I kept the interface in the same file in hopes to "try to follow" that conventions.

But it's really a bad habit IMHO having big files with plenty of classes.